pythonwhat/State.py

```diff
@@ -79,7 +109,26 @@ def get_manual_sigs(self):
 
         return(self.manual_sigs)
 
-    def to_child_state(self, student_subtree, solution_subtree):
+    def build_message(self, tail="", fmt_kwargs=None):
+        if not fmt_kwargs: fmt_kwargs = {}
+        out_list = []
+        # add trailing message to msg list
+        msgs = self.messages[:] + [{'msg': tail or "", 'kwargs':fmt_kwargs}]
+        # format messages in list, by iterating over previous, current, and next message
+        for prev_d, d, next_d in zip([{}, *msgs[:-1]], msgs, [*msgs[1:], {}]):
+            out = d['msg'].format(parent = prev_d.get('kwargs'),
```

Comment:
Just out of curiosity: you're using `.get` instead of `[]` to subset dictionaries, is that right? Is there a reason for that? Is it considered better style?